api/lib/src/lib.rs

```diff
@@ -321,8 +329,8 @@ impl AddressString {
 			.map_err(|_| anyhow!("Failed to parse address"))
 	}
 
-	pub fn from_encoded_address(address: &EncodedAddress) -> Self {
-		Self(address.to_string())
+	pub fn from_encoded_address<T: std::borrow::Borrow<EncodedAddress>>(address: T) -> Self {
+		Self(address.borrow().to_string())
```

Comment:
What's the reason for this change?

Reply:
Just to fix rust analyser complaining about it. If i just removed the reference then clippy would complain about it, so this was the best solution that made both happy.